### ISSUE TITLE
Add history to the password generator

### ISSUE THREAD

**Issue by witcherenjoyer** [keepassxreboot/keepassxc#7297]
## Summary
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Provide a brief overview of what the new feature is all about )
pretty self explanatory, adding a log of passwords that have been generated during the session would be a very useful and good feature. 

## Examples
[NOTE]: # ( Show us a picture or mock-up of your proposal )

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )
Just lost a gpg key because i clicked off the gen and had the password change before i could save it :(

**Comment by droidmonkey** [comment]
The password generator is stateless by design. You could inadvertently leak information, that password history list needs to be stored somewhere. Storing it in memory is an option, but would be cleared on database lock and application close. The standalone password generator will never have a history due to it being outside of an encrypted environment.

**Comment by witcherenjoyer** [comment]
then would it be possible to rework the generator to not generate a new password until explicitly told to do so? what happened to me, is i accidentally clicked on passphrase from password, and it removed my password before it could be saved.

**Comment by droidmonkey** [comment]
Yes that is a better user experience anyway, it is rather unexpected to have the password regenerate when switching tabs.

**Comment by ygrange** [comment]
I just came here because I changed my password, and then accidentally changed the character set after pasting on the website and submitting. I think it is especially confusing that of course since passwords are only shown as dots, you don't see the password change either. I think I'd support having to explicitly refresh the password after changing the charset settings.

**Comment by ccope** [comment]
This exacerbates a bug where I use the KeePassXC browser extension in Firefox to generate a new password, have it fill the form, click the form submit button, and then the "update existing entry or add new" bar appears at the top of the window while the page redirects to the logged in site and then the bar is gone. If the generator saved history in memory until the database gets locked again, this would be a total non-issue.

**Comment by ghost** [comment]
Unfortunately those situations tend to happen to me more often than I would like. I'm hoping this usability issue gets resolved as soon as possible... :P

**Comment by lionkmp** [comment]
To overcome the problem mentioned by OP I **always** have a "TEMP" entry in my Root, where I generate new passwords, save them before using them. I do not ever submit or use a password before saving them to TEMP. (But yes, I also had my trouble first...)

Maybe my workflow could be implemented, by adding a "Save to Generated Passwords" (next to Cancel button) that creates and updates a "hardcoded" Root/"Generated Passwords" (one item, not a folder, the item gets a new password added to its history.) (Or the item could be called "New Passwords" and button "Save to New Passwords" if we see example in #8415 )

(That button would be visible only when not started from an entry.)

**Comment by sjvudp** [comment]
Probably the solution is: "First create and save (or remember) a password, *then* use it." ;-)

**Comment by jrabbit** [comment]
> Probably the solution is: "First create and save (or remember) a password, _then_ use it." ;-)

No.

**Comment by sjvudp** [comment]
> > Probably the solution is: "First create and save (or remember) a password, _then_ use it." ;-)
> 
> No.

But isn't "Just lost a gpg key because i clicked off the gen and had the password change before i could save it :(" just a user error?
The correct usage is to let the password generator set a password and then use that; isn't it?

**Comment by Jookia** [comment]
> This exacerbates a bug where I use the KeePassXC browser extension in Firefox to generate a new password, have it fill the form, click the form submit button, and then the "update existing entry or add new" bar appears at the top of the window while the page redirects to the logged in site and then the bar is gone. If the generator saved history in memory until the database gets locked again, this would be a total non-issue.

This is exactly why I'm afraid to use the browser generator. :(

An in-memory log of passwords would help this.

**Comment by Groxx** [comment]
Could this be saved to the database instead?  It's wild to me that I can generate a password, say "fill", submit it, *and not ever get the "save password" prompt* (because that depends on detecting the right fields) and effectively lose the password entirely without any warning at all.  
I would **much** prefer to have to unlock to generate, or unlock on "fill"/"copy".

**Comment by shaund** [comment]
It'd be really nice if it saved all generated passwords to the database. LastPass does it and I recently recovered an old crypto account where the generated password was never saved to an entry, because I was able to look at my old generated passwords from years ago.

**Comment by murph146** [comment]
Just coming here to say an implementation of the password generator history would be an awesome additional and very common in password managers.
